tests/test_physics/test_ansatz.py

```diff
@@ -155,3 +154,102 @@ def test_PolyExpAnsatz_simplify_v2(Abc):
     assert len(ansatz.A) == 1
     assert len(ansatz.b) == 1
     assert np.allclose(ansatz.c, 2 * c)
+
+
+class TestArrayAnsatz:
+    r"""Tests all algebra related to ArrayAnsatz."""
+
+    def test_ArrayAnsatz_init_(self):
+        r"""Tests that an ArrayAnstaz can be initialized."""
+        array = np.random.random((2, 4, 5))
+        aa = ArrayAnsatz(array=array)
+        assert isinstance(aa, ArrayAnsatz)
+        assert np.allclose(aa.array, array)
+
+    def test_ArrayAnsatz_neg(self):
+        r"""Negates the array inside ArrayAnsatz."""
+        array = np.random.random((2, 4, 5))
+        aa = ArrayAnsatz(array=array)
+        minusaa = -aa
+        assert isinstance(minusaa, ArrayAnsatz)
+        assert np.allclose(minusaa.array, -array)
+
+    def test_ArrayAnsatz_equal(self):
+        r"""Tests the equation of two ArrayAnsatzs."""
+        array = np.random.random((2, 4, 5))
+        aa1 = ArrayAnsatz(array=array)
+        aa2 = ArrayAnsatz(array=array)
+        assert aa1 == aa2
+
+    def test_ArrayAnsatz_addition(self):
+        r"""Tests the correctness of adding two ArrayAnsatzs."""
+        array = np.random.random((2, 4, 5))
+        array2 = np.random.random((4, 4, 5))
+        aa1 = ArrayAnsatz(array=array)
+        aa2 = ArrayAnsatz(array=array2)
+        aa1_add_aa2 = aa1 + aa2
+        assert isinstance(aa1_add_aa2, ArrayAnsatz)
+        assert aa1_add_aa2.array.shape == (8, 4, 5)
+
+    def test_ArrayAnsatz_and(self):
+        r"""Tests the correctness of adding two ArrayAnsatzs."""
```

Comment:
adding? or "tests the ``and`` operator", so that it's different from the one above?

Reply:
and is the outer product like mode 1 and mode 2, add is to add something in the same dimension like mode 1 and mode 1.